dgraph/cmd/alpha/run.go

```diff
@@ -808,9 +808,15 @@ func listenForCorsUpdate(closer *z.Closer) {
 	// Remove uid from the key, to get the correct prefix
 	prefix = prefix[:len(prefix)-8]
 	worker.SubscribeForUpdates([][]byte{prefix}, func(kvs *badgerpb.KVList) {
-		// Last update contains the latest value. So, taking the last update.
-		lastIdx := len(kvs.GetKv()) - 1
-		kv := kvs.GetKv()[lastIdx]
+		// Iterate over kvs to get the KV with the latest version. It is not necessary that the last
+		// KV contain the latest value.
+		var kv badgerpb.KV
+		for _, Kv := range kvs.GetKv() {
+			if Kv.GetVersion() > kv.GetVersion() {
```

Comment:
Check the key as well.

Reply:
Why do we need to check key?

Reply:
I don't think there will be any benefit in comparing the key, unless we suspect that badger might have sent the key which does not match the subscribed prefixes.